serde-jackson/src/main/java/io/micronaut/serde/jackson/JacksonDecoder.java

```diff
@@ -120,11 +124,27 @@ protected boolean getBoolean() throws IOException {
 
     @Override
     protected long getLong() throws IOException {
+        JsonToken jsonToken = parser.currentToken();
+        if (jsonToken == JsonToken.VALUE_NUMBER_INT || jsonToken == JsonToken.VALUE_NUMBER_FLOAT) {
```

Comment:
why is this faster? This is pretty much exactly the code in getValueAsLong, if it doesn't get inlined properly that should be fixed

Reply:
I don't know but somehow it's faster

Reply:
Could maybe check with jitwatch what the generated code looks like? This should be easy to inline, unless there is some polymorphism in the JsonParser impls or something

Reply:
JMH tests got a better performance so I believe it. I can show you how to eproduce it if you want to take a look.

Reply:
yes please, are the jmh test cases around somewhere?

Reply:
We are using this branch https://github.com/graemerocher/java-json-benchmark/tree/micronaut plus I had to add management dependency to bypass some weird error